extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/storage/QuarkusLocalStorageKeyVertxServiceProvider.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.vertx.runtime.storage;
+
+import io.quarkus.arc.InjectableContext;
+import io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle;
+import io.vertx.core.impl.VertxBuilder;
+import io.vertx.core.spi.VertxServiceProvider;
+import io.vertx.core.spi.context.storage.ContextLocal;
+
+/**
+ * This provider exists with the sole purpose of reliably get the optimized local keys created before
+ * the Vertx instance is created.
+ */
+public class QuarkusLocalStorageKeyVertxServiceProvider implements VertxServiceProvider {
```

Comment:
@geoand @cescoffier I don't like to make spi loading to happen just for this TBH and I could, maybe, hack the `Vertx` recorder to just perform the init at the right time, but IDK what's the cost of this, usually. Any suggestion?

Reply:
The question is: can we preload this at build time (static init). It would be interesting to do that and just use the result at runtime. I don't believe there is a good reason to change the SPI at runtime.

Reply:
I think we can skip the service loading in this case and just use this class,  no?

Reply:
Yeah, I don't believe we should allow any other implementation.

Reply:
thanks both, yeah, I'm not very happy about it: so i can just have a not vertx service provider, but an holder class which is going to be initialized in the vertx recorder before vertx instances are allocated, wdyt?

Reply:
Sounds fine to me